src/main/java/racingcar/utils/validator/CarValidator.java

```diff
@@ -0,0 +1,56 @@
+package racingcar.utils.validator;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import racingcar.domain.car.Car;
+import racingcar.utils.message.ErrorMessage;
+
+public class CarValidator {
+    private static final int MINIMUM_CARS_SIZE = 2;
+    private static final int MAXIMUM_CARS_SIZE = 5;
+    private static final String NAME_REGEX = "^[a-zA-Z가-힣]{1,5}$";
+    private static final Pattern pattern = Pattern.compile(NAME_REGEX);
+
+    private CarValidator() {}
+
+    public static void validateCars(List<Car> cars) {
+        validateCarSize(cars);
+        validateDuplicateName(cars);
+    }
+
+    public static void validateName(String name) {
+        if (isNotMatchValidNameCondition(name)) {
+            throw new IllegalArgumentException(ErrorMessage.INVALID_NAME_CONDITION.getMessage());
+        }
+    }
+
+    private static void validateCarSize(List<Car> cars) {
+        if (isOverValidCarSize(cars.size())) {
+            throw new IllegalArgumentException(ErrorMessage.INVALID_NAME_NUMBER.getMessage());
+        }
+    }
+
+    private static void validateDuplicateName(List<Car> cars) {
+        if (isContainDuplicateName(cars)) {
+            throw new IllegalArgumentException(ErrorMessage.DUPLICATE_CAR_NAME.getMessage());
+        }
+    }
+
+    private static boolean isOverValidCarSize(int size) {
+        return size < MINIMUM_CARS_SIZE || MAXIMUM_CARS_SIZE < size;
+    }
+
+    private static boolean isContainDuplicateName(List<Car> cars) {
+        List<Car> target = cars.stream()
+                .map(Car::getName)
+                .distinct()
+                .map(name -> new Car(name))
+                .collect(Collectors.toList());
```

Comment:
스트림을 사용하는 것 보다 HashSet을 통해서 비교하면 더 가독성이 좋은 코드가 만들어질 것 같아요 :)

```suggestion
        List<Car> target = new HashSet(car);
```

Reply:
처음에 Set을 이용하려고 했지만, 객체의 동등성을 검사하는 방법을 생각하지 못했습니다.
돌이켜보니 hashcode 와 equals를 재정의하면 좋겠다는 생각이 듭니다!